### ISSUE TITLE
FMX, "Enable Skia" makes custom fonts unavailable

### ISSUE THREAD

**Issue by andrea-magni** [skia4delphi/skia4delphi#153]
Hi, congratulations for your project.

I have an issue that can be reproduced following these steps (I'm using Delphi 11.2 Alexandria, recently released):

1.  Create a new empty FMX application
2. Add a TLabel and use a custom font
3. Add the custom font file to Deployment (for Android)
4. Run the application --> everything works fine (on Android specifically, of course)
5. Click on the Project manager and "Enable Skia"
6. Run again the project and the custom font is gone

I am attaching a working FMX application. If you "Enable Skia" on this project, you'll see the problem.
This is blocking for me and my FMXER project and I would say this was not happening with Delphi 11.1 (but this may not be the only factor).

I would guess it has something to do with the .deployproj file.

Thanks in advance for any hint/suggestion.
[Test_MD_No_Skia.zip](https://github.com/skia4delphi/skia4delphi/files/9547168/Test_MD_No_Skia.zip)

**Comment by andrea-magni** [comment]
Thanks to the kind help of Vinicius and other MVPs, I managed to workaround the issue using:

```
initialization
  TSkTypefaceManager.RegisterTypeface(TPath.Combine(TPath.GetDocumentsPath, 'Material Design Icons Desktop.ttf'))
```
If I am not wrong, this happens because Skia replaces the FMX font management (that would register fonts on demand, looking at certain paths) so you need to register the custom fonts against Skia management.

Thanks a lot, I am updating my projects.

Feel free to close this issue, thanks!

**Comment by viniciusfbb** [comment]
Although there are still open issues about this in the quality center, I just confirmed in 11.2 that in fact the custom fonts on Android that are in the 'assets\internals' folder (TPath.GetDocumentsPath) are automatically loaded by the FMX default Canvas. You need to implement the same behavior in Skia Canvas.

One note about FMX's default Canvas behavior is that it only looks for the .ttf/.otf in 'assets\internals'. The search is not done in subdirectories.

[CustomFontAndroid.zip](https://github.com/skia4delphi/skia4delphi/files/9592068/CustomFontAndroid.zip)

**Comment by viniciusfbb** [comment]
Any news?

**Comment by viniciusfbb** [comment]
Fixed in [new release v4.0.0](https://github.com/skia4delphi/skia4delphi/releases)
